### PR TITLE
Ensure generated formula code handles prev correctly for i,ii in ticking tables. Fixes #544

### LINKED ISSUE #544: Formula code uses wrong index for prev methods on i and ii for ticking tables.
Relatively easy to repro w/ my FullAdder demo. (Can give a demo branch later.)

Added some additional non-null checks to trace the culprit, here it is:

```
java.lang.NullPointerException: null
        at java.util.Objects.requireNonNull(Objects.java:203)
        at io.deephaven.db.v2.utils.TreeIndex.<init>(TreeIndex.java:100)
        at io.deephaven.db.v2.utils.TreeIndex.invert(TreeIndex.java:303)
        at io.deephaven.db.v2.utils.SortedIndex.invert(SortedIndex.java:105)
        at io.deephaven.temp.cm1760100514987228486v52_0.Formula.fillChunkHelper(Formula.java:150)
        at io.deephaven.temp.cm1760100514987228486v52_0.Formula.fillPrevChunk(Formula.java:142)
        at io.deephaven.db.v2.select.Formula.getPrevChunk(Formula.java:166)
        at io.deephaven.db.v2.sources.ViewColumnSource.getPrevChunk(ViewColumnSource.java:234)
        at io.deephaven.db.v2.by.ChunkedOperatorAggregationHelper.fetchValues(ChunkedOperatorAggregationHelper.java:1769)
        at io.deephaven.db.v2.by.ChunkedOperatorAggregationHelper.doNoKeyUpdate(ChunkedOperatorAggregationHelper.java:1676)
        at io.deephaven.db.v2.by.ChunkedOperatorAggregationHelper.doNoKeyRemoval(ChunkedOperatorAggregationHelper.java:1571)
        at io.deephaven.db.v2.by.ChunkedOperatorAggregationHelper.access$2600(ChunkedOperatorAggregationHelper.java:38)
        at io.deephaven.db.v2.by.ChunkedOperatorAggregationHelper$2.onUpdate(ChunkedOperatorAggregationHelper.java:1468)
        at io.deephaven.db.v2.InstrumentedShiftAwareListener$Notification.lambda$run$0(InstrumentedShiftAwareListener.java:36)
        at io.deephaven.db.v2.InstrumentedListenerBase$NotificationBase.doRunInternal(InstrumentedListenerBase.java:223)
        at io.deephaven.db.v2.InstrumentedListenerBase$NotificationBase.doRun(InstrumentedListenerBase.java:205)
        at io.deephaven.db.v2.InstrumentedShiftAwareListener$Notification.run(InstrumentedShiftAwareListener.java:36)
        at io.deephaven.db.tables.live.LiveTableMonitor.runNotification(LiveTableMonitor.java:1308)
        at io.deephaven.db.tables.live.LiveTableMonitor.access$1100(LiveTableMonitor.java:70)
        at io.deephaven.db.tables.live.LiveTableMonitor$QueueNotificationProcessor.doWork(LiveTableMonitor.java:1457)
        at io.deephaven.db.tables.live.LiveTableMonitor.flushNormalNotificationsAndCompleteCycle(LiveTableMonitor.java:1192)
        at io.deephaven.db.tables.live.LiveTableMonitor.flushNotificationsAndCompleteCycle(LiveTableMonitor.java:1141)
        at io.deephaven.db.tables.live.LiveTableMonitor.lambda$doRefresh$14(LiveTableMonitor.java:1717)
        at io.deephaven.util.locks.FunctionalLock.doLocked(FunctionalLock.java:29)
        at io.deephaven.db.tables.live.LiveTableMonitor.doRefresh(LiveTableMonitor.java:1707)
        at io.deephaven.db.tables.live.LiveTableMonitor.drainSingleUpdateQueue(LiveTableMonitor.java:1668)
        at io.deephaven.db.tables.live.LiveTableMonitor.waitForEndTime(LiveTableMonitor.java:1631)
        at io.deephaven.db.tables.live.LiveTableMonitor.waitForNextCycle(LiveTableMonitor.java:1613)
        at io.deephaven.db.tables.live.LiveTableMonitor.refreshTablesAndFlushNotifications(LiveTableMonitor.java:1587)
        at io.deephaven.db.tables.live.LiveTableMonitor.access$300(LiveTableMonitor.java:70)
        at io.deephaven.db.tables.live.LiveTableMonitor$1.run(LiveTableMonitor.java:195)
```

### PR DESCRIPTION
**Pull request by jcferretti**
New unit test case that fails without this change added to QueryTableTest.
TestFormulaColumnGenerator ran and FormulaSample was generated, and tested.

### CODE REVIEW

**Comment by devinrsmith** [comment]
My demo is working w/ this fix.

**Review by kosak — APPROVED** [review]
This is a beautiful change. Thank you for working on this!

**Review by rcaudy — CHANGES_REQUESTED** [review]
Please use try-with-resources for the 3 indexes we touch.

File: DB/src/test/java/io/deephaven/db/v2/select/FormulaSample.java [hunk 1]
```diff
@@ -142,24 +144,25 @@ public void fillChunk(final FillContext __context, final WritableChunk<? super A
...
-        fillChunkHelper(__typedContext, __destination, __orderedKeys, __chunk__col__II, __chunk__col__I);
+        fillChunkHelper(true, __typedContext, __destination, __orderedKeys, __chunk__col__II, __chunk__col__I);
     }
 
-    private void fillChunkHelper(final FormulaFillContext __context,
+    private void fillChunkHelper(final boolean __usePrev, final FormulaFillContext __context,
             final WritableChunk<? super Attributes.Values> __destination,
             final OrderedKeys __orderedKeys, LongChunk<? extends Attributes.Values> __chunk__col__II, IntChunk<? extends Attributes.Values> __chunk__col__I) {
         final WritableLongChunk<? super Attributes.Values> __typedDestination = __destination.asWritableLongChunk();
+        final Index inverted = (__usePrev ? __index.getPrevIndex() : __index).invert(__orderedKeys.asIndex());
```
[the remark this answers is not in the document]
Comment (jcferretti) [reply DB/src/test/java/io/deephaven/db/v2/select/FormulaSample.java]: Done, no need to try-with-resources.

File: DB/src/main/java/io/deephaven/db/v2/select/DhFormulaColumn.java [hunk 2]
```diff
@@ -497,17 +510,22 @@ private CodeGenerator generateFillChunk(boolean usePrev) {
...
-                        CodeGenerator.optional("maybeCreateI",
-                                "__context.__iChunk.setSize(0);",
-                                "__index.invert(__orderedKeys.asIndex()).forAllLongs(l -> __context.__iChunk.add(__intSize(l)));"
-                        ),
-                        CodeGenerator.optional("maybeCreateII",
-                                "__index.invert(__orderedKeys.asIndex()).fillKeyIndicesChunk(__context.__iiChunk);"
-                        ),
+                        CodeGenerator.optional("maybeCreateIOrII",
+                                "try (final Index prev = __usePrev ? __index.getPrevIndex() : null)", CodeGenerator.block(
+                                        "try (final Index inverted = ((prev != null) ? prev : __index).invert(__orderedKeys.asIndex()))", CodeGenerator.block(
```
Comment (rcaudy) [inline DB/src/main/java/io/deephaven/db/v2/select/DhFormulaColumn.java:519]: No need to nest these. Use a semicolon to put them in the same block.

**Review by rcaudy — DISMISSED** [review]

File: DB/src/main/java/io/deephaven/db/v2/select/DhFormulaColumn.java (hunk 2 above)
Comment (jcferretti) [reply DB/src/main/java/io/deephaven/db/v2/select/DhFormulaColumn.java]: That was intentional.  Given the little control we get from the CodeGenerator, I like the readability/indentation result of what I had better.  But, suit yourself, I don't care enough to argue.

Comment (rcaudy) [reply DB/src/main/java/io/deephaven/db/v2/select/DhFormulaColumn.java]: I don't, either. If the code gen makes the other way look bad, keep this.

**Review by rcaudy — APPROVED** [review]

**Pull request merged**